pkg/dependency/dotnet_runtime_test.go

```diff
@@ -257,6 +257,7 @@ func testDotnetRuntime(t *testing.T, when spec.G, it spec.S) {
 				SHA:             "some-sha256",
 				ReleaseDate:     "2020-02-20T00:00:00Z",
 				DeprecationDate: "2050-02-20T00:00:00Z",
+				CPE:             "cpe:2.3:a:microsoft:.net_core:2.0.1:*:*:*:*:*:*:*",
```

Comment:
It seems like starting with .Net 5.0 the CPE notation drops the `_core` (see the CPEs at the bottom of [this CVE](https://nvd.nist.gov/vuln/detail/CVE-2021-26701)). It doesn't look like this code accounts for that difference.

Also, based on that CVE it seems like after 5.0 we shouldn't include the patch number in the version either.

Reply:
Hmm, that's an interesting one. I see the CPE on that CVE but searching for [`cpe:2.3:a:microsoft:.net`](https://nvd.nist.gov/products/cpe/search/results?namingFormat=2.3&keyword=cpe%3A2.3%3Aa%3Amicrosoft%3A.net) doesn't show any results.

Reply:
When I searched with `*`s (e.g. [`cpe:2.3:a:microsoft:.net:*:*:*`](https://nvd.nist.gov/vuln/search/results?form_type=Basic&results_type=overview&query=cpe%3A2.3%3Aa%3Amicrosoft%3A.net%3A*%3A*%3A*&search_type=all)) I was able to find results

Reply:
Ah it looks like there are two separate searches - one for finding CPEs and one for finding CVEs. But even if there are no CPEs right now it does seem right to use `.net` for 5.0+ like you said.

Reply:
I took another look and I think that the CVE you linked to actually has the wrong CPE. Notice that if you expand `Show Matching CPE(s)` for all three CPEs, the `.net` one shows `No Matching CPE(s) found in CPE Dictionary` while the other two have matching CPEs.

Also, searching for [`cpe:2.3:a:microsoft:.net_core:5.0.0:*:*:*:*:*:*:*`](https://nvd.nist.gov/products/cpe/search/results?namingFormat=2.3&keyword=cpe%3A2.3%3Aa%3Amicrosoft%3A.net_core%3A5.0.0%3A*%3A*%3A*%3A*%3A*%3A*%3A*) and [`cpe:2.3:a:microsoft:.net_core:5.0:*:*:*:*:*:*:*`](https://nvd.nist.gov/products/cpe/search/results?namingFormat=2.3&keyword=cpe%3A2.3%3Aa%3Amicrosoft%3A.net_core%3A5.0%3A*%3A*%3A*%3A*%3A*%3A*%3A*) shows matches but [`cpe:2.3:a:microsoft:.net:5.0:*:*:*:*:*:*:*`](https://nvd.nist.gov/products/cpe/search/results?namingFormat=2.3&keyword=cpe%3A2.3%3Aa%3Amicrosoft%3A.net%3A5.0%3A*%3A*%3A*%3A*%3A*%3A*%3A*) and [`cpe:2.3:a:microsoft:.net:5.0.0:*:*:*:*:*:*:*`](https://nvd.nist.gov/products/cpe/search/results?namingFormat=2.3&keyword=cpe%3A2.3%3Aa%3Amicrosoft%3A.net%3A5.0.0%3A*%3A*%3A*%3A*%3A*%3A*%3A*) have no matches.

Reply:
Fixed in latest commit